homeassistant/components/rdw/binary_sensor.py

```diff
@@ -0,0 +1,103 @@
+"""Support for RDW binary sensors."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Callable
+
+from vehicle import Vehicle
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_PROBLEM,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import CONF_LICENSE_PLATE, DOMAIN, ENTRY_TYPE_SERVICE
+
+
+@dataclass
+class RDWBinarySensorEntityDescriptionMixin:
+    """Mixin for required keys."""
+
+    is_on_fn: Callable[[Vehicle], bool | None]
+
+
+@dataclass
+class RDWBinarySensorEntityDescription(
+    BinarySensorEntityDescription, RDWBinarySensorEntityDescriptionMixin
+):
+    """Describes RDW binary sensor entity."""
+
+
+BINARY_SENSORS: tuple[RDWBinarySensorEntityDescription, ...] = (
+    RDWBinarySensorEntityDescription(
+        key="liability_insured",
+        name="Liability Insured",
+        icon="mdi:shield-car",
+        is_on_fn=lambda vehicle: vehicle.liability_insured,
+    ),
+    RDWBinarySensorEntityDescription(
+        key="pending_recall",
+        name="Pending Recall",
+        device_class=DEVICE_CLASS_PROBLEM,
+        is_on_fn=lambda vehicle: vehicle.pending_recall,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up RDW binary sensors based on a config entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        RDWBinarySensorEntity(
+            coordinator=coordinator,
+            license_plate=entry.data[CONF_LICENSE_PLATE],
+            description=description,
+        )
+        for description in BINARY_SENSORS
+        if description.is_on_fn(coordinator.data) is not None
+    )
+
+
+class RDWBinarySensorEntity(CoordinatorEntity, BinarySensorEntity):
+    """Defines an RDW binary sensor."""
+
+    entity_description: RDWBinarySensorEntityDescription
+
+    def __init__(
+        self,
+        *,
+        coordinator: DataUpdateCoordinator,
+        license_plate: str,
+        description: RDWBinarySensorEntityDescription,
+    ) -> None:
+        """Initialize RDW binary sensor."""
+        super().__init__(coordinator=coordinator)
+        self.entity_description = description
+        self._attr_unique_id = f"{license_plate}_{description.key}"
+
+        self._attr_device_info = DeviceInfo(
+            entry_type=ENTRY_TYPE_SERVICE,
+            identifiers={(DOMAIN, f"{license_plate}")},
```

Comment:
No need to put in string formatting.
```suggestion
            identifiers={(DOMAIN, license_plate)},
```